src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -201,26 +206,29 @@ default ClassDesc arrayType() {
      * @jvms 4.4.1 The CONSTANT_Class_info Structure
      */
     default ClassDesc arrayType(int rank) {
-        int netRank;
-        if (rank <= 0) {
+        if (rank <= 1) {
+            if (rank == 1) {
+                return arrayType();
+            }
             throw new IllegalArgumentException("rank " + rank + " is not a positive value");
         }
         String desc = descriptorString();
-        try {
-            int currentDepth = arrayDepth(desc);
-            netRank = Math.addExact(currentDepth, rank);
-            if (netRank > MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
-                throw new IllegalArgumentException("rank: " + netRank +
-                                                   " exceeds maximum supported dimension of " +
-                                                   MAX_ARRAY_TYPE_DESC_DIMENSIONS);
-            }
-        } catch (ArithmeticException ae) {
-            throw new IllegalArgumentException("Integer overflow in rank computation");
+        long currentDepth = arrayDepth(desc);
+        long netRank = currentDepth + rank;
+        if (netRank > MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
+            throw new IllegalArgumentException("rank: " + netRank +
+                    " exceeds maximum supported dimension of " +
+                    MAX_ARRAY_TYPE_DESC_DIMENSIONS);
+        }
+        var sb = new StringBuilder(desc.length() + rank);
+        for (int i = 0; i < rank; i++) {
+            sb.append('[');
         }
```

Comment:
Would `sb.repeat("[", rank);` be better here?

Also for parity with `arrayType()` I recommend moving this building code to after the void check.

Reply:
Oh, that's new (in 21). I was comparing against `append("[".repeat(rank))` which had a cost here. `sb.repeat('[', rank)` is in the same ballpark and simpler. Fixing.

```
Benchmark                                       Mode  Cnt     Score     Error   Units
ClassDescMethods.arrayType2                     avgt   15    31,568 ±   2,108   ns/op
ClassDescMethods.arrayType2:gc.alloc.rate       avgt   15  7758,056 ± 474,378  MB/sec
ClassDescMethods.arrayType2:gc.alloc.rate.norm  avgt   15   256,000 ±   0,001    B/op
```

I'll align the code to construct `newDesc` before the void check in both methods.

Reply:
Didn't realize the char version actually takes an int codepoint :smile: so I suggested the String one.